### PR TITLE
[notifications] warn about expo go usage

### PR DESCRIPTION
**Pull request by vonovak**
# Why

notifications are one of those things that are best used with development builds, and we should warn developers when they are using the package with expo go

# How

add a warning in the root.

I'm thinking of where this should be added. It can be as it is in the PR but that might be a little too annoying ~(it may display often due to hot reloading) - OTOH I think we want it to be a bit annoying~ It's only being displayed once.

Otherwise we can also put it in some selected exports (like requesting a push token or presenting a notification) but I think this might be best.

# Test Plan

tested locally

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by vonovak** [comment]
* **#31573** <a href="https://app.graphite.dev/github/pr/expo/expo/31573?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @vonovak and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/expo-notifications/src/index.ts [hunk 1]
```diff
@@ -1,3 +1,12 @@
+import { isRunningInExpoGo } from 'expo';
+
+if (isRunningInExpoGo()) {
+  const message =
+    '`expo-push-notifications` functionality is not fully supported in Expo Go:\n' +
```
Comment (tsapeta) [inline packages/expo-notifications/src/index.ts:5]:
```suggestion
    '`expo-notifications` functionality is not fully supported in Expo Go:\n' +
```

File: packages/expo-notifications/CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,8 @@
 
 ### 🎉 New features
 
+- warn about usage with Expo Go ([#31573](https://github.com/expo/expo/pull/31573) by [@vonovak](https://github.com/vonovak))
```
Comment (tsapeta) [inline packages/expo-notifications/CHANGELOG.md:11]: It's not a new feature 😅 Let's move it to `Others` category and maybe rephrase to something like
```suggestion
- Warn about limited support in Expo Go ([#31573](https://github.com/expo/expo/pull/31573) by [@vonovak](https://github.com/vonovak))
```
which seems to be more meaningful

**Review by tsapeta — APPROVED** [review]

**Review by douglowder — APPROVED** [review]
